__tests__/app.test.js

```diff
@@ -111,4 +112,192 @@ describe('GET requests', () => {
+        });
+        test('404 - article_id not found ', () => {
+            return request(app)
+            .get('/api/articles/999/comments')
+            .expect(404)
+            .then(({body}) => {
+                expect(body.msg).toBe('No articles found for article_id: 999')
+            })
+        })
+        test('400 - bad request', () => {
+            return request(app)
+            .get('/api/articles/itchy/comments')
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe('invalid request')
+            })
+        })
+    });
+
+
 });
+
+describe('POST', () => {
+    
+    describe('POST /api/articles/:article_id/comments', () => {
+        
+        test('201 - responds with the posted comment', () => {
+            const newComment = {body: 'Wonderful, simply wonderful', username: 'rogersop'}
+            return request(app)
+            .post('/api/articles/2/comments')
+            .send(newComment)
+            .expect(201)
+            .then(({ body }) => {
+                expect(body.comment).toMatchObject({
+                    comment_id: 19,
+                    body: 'Wonderful, simply wonderful',
+                    article_id: 2,
+                    author: 'rogersop',
+                    votes: 0
+                })
+                expect(body.comment).toHaveProperty("created_at")
+            })
+        });
+        test('404 - article_id not found', () => {
+            const newComment = {body: 'Wonderful, simply wonderful', username: 'rogersop'}
+            return request(app)
+            .post('/api/articles/999/comments')
+            .send(newComment)
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe('provided key not found')
+            })
+        });
+        test('404 - user not found', () => {
+            const newComment = {body: 'Wonderful, simply wonderful', username: 'unknownUser'}
+            return request(app)
+            .post('/api/articles/2/comments')
+            .send(newComment)
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe('provided key not found')
+            })
+        });
+        test('400 - comment body not provided', () => {
+            const malformed = {username: 'rogersop'}
+            return request(app)
+            .post('/api/articles/2/comments')
+            .send(malformed)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe('Please provide comment body.')
+            })
+        })
+        test('400 - comment username not provided', () => {
+            const malformed2 = {body: 'Great stuff'}
+            return request(app)
+            .post('/api/articles/2/comments')
+            .send(malformed2)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe('Please provide username.')
+            })
+        })
+
```

Comment:
Great work on these tests! A couple more to include: a test for extra properties being ignored i.e. we send body and username but also a third key-value pair that is ignored. We should also test for something like /api/articles/not-an-id/comments which would be a 400 - this will be a psql error